### PR TITLE
add wmt14 and sampler

### PR DESCRIPTION
**Pull request by szhengac**
## Description ##
add wmt14 and sampler

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] WMT14 and tests
- [x] FixedSampler update to use average bucket length

### CODE REVIEW

File: gluonnlp/data/sampler.py [hunk 1]
```diff
@@ -116,9 +116,9 @@ class FixedBucketSampler(Sampler):
         smaller buckets.
     shuffle : bool, default False
         Whether to shuffle the batches.
-    batch_type : str, default 'sentence'
-        'sentence': each batch contains batch_size sentences, number of words varies.
-        'word': each batch contains batch_size tokens, number of sentences varies.
+    use_average_length : bool, default False
+        False: each batch contains batch_size sequences, number of sequence elements varies.
+        True: each batch contains batch_size elements, number of sequences varies. In this case, ratio is disabled.
```
Comment (szha) [inline gluonnlp/data/sampler.py]: `ratio` option is ignored.

File: gluonnlp/data/sampler.py [hunk 2]
```diff
@@ -207,16 +207,14 @@ def __init__(self, lengths, batch_size, num_buckets=10, bucket_keys=None,
                                    if len(sample_ids) > 0]
         scale_up_keys = [key if self._single_element else sum(key) for key in self._bucket_keys]
         max_scale_up_key = max(scale_up_keys)
-        if batch_type == 'sentence':
+        if ~use_average_length:
```
Comment (szha) [inline gluonnlp/data/sampler.py]: `if not use_avg_length`

Comment (szha) [reply gluonnlp/data/sampler.py]: also, the scale_up_keys only needs to be calculated in this case, so you can move those logic into this `if` block.

**Comment by mli** [comment]
Job PR-135/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-135/5/index.html

File: gluonnlp/data/sampler.py [hunk 3]
```diff
@@ -192,11 +206,16 @@ def __init__(self, lengths, batch_size, num_buckets=10, bucket_keys=None,
...
-                                            * self._ratio * batch_size), batch_size)
-                                    for scale_up_key in scale_up_keys]
+        if not use_average_length:
+            scale_up_keys = [key if self._single_element else sum(key) for key in self._bucket_keys]
+            max_scale_up_key = max(scale_up_keys)
+            self._bucket_batch_sizes = [max(int(max_scale_up_key / float(scale_up_key)
+                                                * self._ratio * batch_size), batch_size)
+                                        for scale_up_key in scale_up_keys]
+        else:
+            bucket_average_lengths = _bucket_average_lengths(self._bucket_sample_ids, self._lengths)
```
Comment (szha) [inline gluonnlp/data/sampler.py]: give a warning here when ratio is specified

**Comment by mli** [comment]
Job PR-135/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-135/6/index.html

**Comment by mli** [comment]
Job PR-135/9 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-135/9/index.html

**Pull request merged**